Assignment2Q1.java

```diff
@@ -0,0 +1,97 @@
+import java.util.LinkedList;
+
+/**
+ * Created by Alexandra on 23.05.2017.
+ */
+public class Assignment2Q1 {
+
+    public static class TreeNode <T> {
+        private TreeNode<T> left;
+        private TreeNode<T> right;
+        private T value;
+
+        public TreeNode(TreeNode<T> left, TreeNode<T> right, T value) {
+            if (value == null) {
+                throw new IllegalArgumentException();
+            }
+            this.left = left;
+            this.right = right;
+            this.value = value;
+        }
+
+        public TreeNode<T> getLeft() {
+            return left;
+        }
+
+        public TreeNode<T> getRight() {
+            return right;
+        }
+
+        public T getValue() {
+            return value;
+        }
+
+        public String toString() {
+            return value.toString() + '\t';
+        }
+
+        public void printAncestors(T value) {
+            if (this.value.equals(value)) {
+                System.out.println("No ancestors."); // this node is the root node
+                return;
+            }
+            LinkedList<TreeNode<T>> ancestors = new LinkedList<>(); // stack of ancestors

```

Comment:
FYI: If you are using like a Stack, then no need to use a LinkedList, you could use an ArrayList (and add and remove at the end). A LinkedList has more memory and CPU overhead. If you wanted a Queue rather than a stack, then a linked list would be appropriate.